src/background/tab-state.js

```diff
@@ -160,8 +164,23 @@ export default function TabState(initialState, onchange) {
   this.updateAnnotationCount = async function (tabId, tabUrl) {
     const INITIAL_WAIT_MS = 1000;
     const MAX_WAIT_MS = 3000;
+    const EXPIRATION_CACHE_MS = 3000;
 
     const pendingRequest = pendingAnnotationCountRequests.get(tabId);
+
+    let url;
+    try {
+      url = uriInfo.shouldQueryUri(tabUrl);
+    } catch {
```

Comment:
This generic catch could potentially mask unexpected errors. Looking at the current implementation of `shouldQueryUri` it will currently throw either `BadgeUriError` if the URI is blocked or `TypeError` if the URI failed to parse. If `shouldQueryUri` caught the latter exception and converted it to a `BadgeUriError` then you could catch just `BadgeUriError` exceptions here and re-throw other, unexpected exceptions.

Reply:
I can't think of a scenario where you want to act differently to the `BadgeUrlError` or  the `TypeError` (from failing to parse the URL).